projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -18,3 +20,180 @@ function jetpack_blogging_prompts_add_meta_data( $keys ) {
 	$keys[] = '_jetpack_blogging_prompt_key';
 	return $keys;
 }
+
+add_filter( 'rest_api_allowed_public_metadata', 'jetpack_blogging_prompts_add_meta_data' );
+
+/**
+ * Sets up a new post as an answer to a blogging prompt.
+ *
+ * Called on `wp_insert_post` hook.
+ *
+ * @param int $post_id ID of post being inserted.
+ * @return void
+ */
+function jetpack_setup_blogging_prompt_response( $post_id ) {
+	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	$prompt_id = isset( $_GET['answer_prompt'] ) && absint( $_GET['answer_prompt'] ) ? absint( $_GET['answer_prompt'] ) : false;
+
+	if ( ! jetpack_is_new_post_screen() || ! $prompt_id ) {
```

Comment:
In the calypso editor, this is exiting early and not setting metadata because `jetpack_is_new_post_screen` is only checking for the wp-admin screen.

Reply:
I'm not seeing that, it logs true for me (tested a Simple site), which makes sense because it is the correct screen within the wp-admin iframe.

However, the query param does need to be passed through to the iframed editor. PR for that at https://github.com/Automattic/wp-calypso/pull/70052

Reply:
I tested and merged Automattic/wp-calypso#70052